src/plugins/plot/spectralAggregatePlot/SpectralAggregateView.vue

```diff
@@ -21,15 +21,314 @@
+            if (!spectralAggregateValue) {
+                this.addSpectralAggregateType(key, telemetryObject.name, formattedTimestamp, color);
+            }
+
+            if (data.message) {
+                this.openmct.notifications.alert(data.message);
+            }
+
+            let xValues = [];
+            let yValues = [];
+
+            //populate X and Y values for plotly
+            axisMetadata.xAxisMetadata.forEach((metadata) => {
+                xValues.push(metadata.name);
+                if (data[metadata.key]) {
+                    yValues.push(data[metadata.key]);
+                } else {
+                    yValues.push('');
+                }
+            });
+
+            const trace = {
+                key,
+                name: telemetryObject.name,
+                x: xValues,
+                y: yValues,
+                text: yValues.map(String),
+                xAxisMetadata: axisMetadata.xAxisMetadata,
+                yAxisMetadata: axisMetadata.yAxisMetadata,
+                type: 'bar'
+            };
+
+            this.addTrace(trace, key);
+        },
+        requestDataFor(telemetryObject) {
+            const axisMetadata = this.getAxisMetadata(telemetryObject);
+            this.openmct.telemetry.request(telemetryObject, this.getOptions(telemetryObject))
+                .then(data => {
+                    data.forEach((datum) => {
+                        this.processData(telemetryObject, datum, axisMetadata);
+                    });
+                });
+        },
+        subscribeToObject(telemetryObject) {
+            const key = objectUtils.makeKeyString(telemetryObject.identifier);
+            const found = Object.values(this.subscriptions).findIndex(objectKey => objectKey === key);
+            if (found > -1) {
+                this.subscriptions[found].unsubscribe();
+                delete this.subscriptions[found];
+            }
+
+            const options = this.getOptions(telemetryObject);
+            const axisMetadata = this.getAxisMetadata(telemetryObject);
+            const unsubscribe = this.openmct.telemetry.subscribe(telemetryObject,
+                data => this.processData(telemetryObject, data, axisMetadata)
+                , options);
+
+            this.subscriptions.push({
+                key,
+                unsubscribe
+            });
+        },
+        subscribeToAll() {
+            this.colorPalette.reset();
+            const telemetryObjects = Object.values(this.telemetryObjects);
+            telemetryObjects.forEach(this.subscribeToObject);
+        },
+        updateDomainObject(newDomainObject) {
+            this.currentDomainObject = newDomainObject;
+        },
+        updateTraceVisibility() {
+            // We create a copy here to improve performance since visibleData - a computed property - is dependent on this.trace.
+            this.trace = this.trace.map((currentTrace, index) => {
```

Comment:
We don't need this functionality, so we should remove this. I'll create an issue for this for later - we can leave this in for now.